maps/biter_battles_v2/functions.lua

```diff
@@ -454,6 +409,193 @@ function Public.map_intro_click(player, element)
+		end
+		local evo_gain = (amount_of_food_this_iteration * diminishing_modifier)
+		evo = evo + evo_gain
+
+		--ADD INSTANT THREAT
+		local diminishing_modifier = 1 / (0.2 + (e2 * 0.016))
+		threat = threat + (amount_of_food_this_iteration * diminishing_modifier)
+
+		food = food - amount_of_food_this_iteration
+	end
+	-- Calculates reanimation chance. This value is normalized onto
+	-- maximum re-animation threshold. For example if real evolution is 150
+	-- and max is 350, then 150 / 350 = 42% chance.
+	local reanim_chance = math_floor(math.max(evo - 1.0, 0) * 100.0)
+	reanim_chance = reanim_chance / max_reanim_thresh * 100
+	reanim_chance = math.min(math_floor(reanim_chance), 90.0)
+
+	threat = threat * get_instant_threat_player_count_modifier(current_player_count)
+	-- Adjust threat for revive.
+	-- Note that the fact that this is done at the end, after reanim_chance is calculated
+	-- is what gives a bonus to large single throws of science rather than many smaller
+	-- throws (in the case where final evolution is above 100%). Specifically, all of the
+	-- science thrown gets the threat increase that would be used for the final evolution
+	-- value.
+	if reanim_chance > 0 then
+		threat = threat * (100 / (100.001 - reanim_chance))
+	end
+
+	return {
+		evo_increase = evo - initial_evo,
+		threat_increase = threat,
+		reanim_chance = reanim_chance
+	}
+end
+
+-- Player can be nil
+function Public.calc_send_command(params, global_passed_in, player_count, player)
+	if params == nil then
+		params = ""
+	end
+	local difficulty = global_passed_in.difficulty_vote_value * 100
+	local evo = nil
+	local print_help = false
+	local error_msg
+	local flask_color
+	local flask_count
+	local force_to_send_to
+	local help_text = "\nUsage: /calc-send evo=20.0 difficulty=30 players=4 color=green count=1000" ..
+		"\nUsage: /calc-send force=north color=white count=1000"
+	if player and player.force.name == "north" then
+		force_to_send_to = "south"
+	elseif player and player.force.name == "south" then
+		force_to_send_to = "north"
+	end
+	-- indexed by strings like "automation-science-pack"
+	local foods = {}
+	for param in string.gmatch(params, "([^%s]+)") do
+		local k, v = string.match(param, "^(%w+)=([%w%p]+)$")
+		if k and v then
+			if k == "force" then
+				if v == "n" or v == "nth" or v == "north" then
+					force_to_send_to = "north"
+				elseif v == "s" or v == "sth" or v == "south" then
+					force_to_send_to = "south"
+				else
+					error_msg = "Invalid force"
+				end
+			elseif k == "evo" then
+				evo = tonumber(v)
+				if evo == nil or evo < 0 or evo > 100000 then
+					error_msg = "Invalid evo"
+				end
+			elseif k == "difficulty" then
+				difficulty = tonumber(v)
+				if difficulty == nil or difficulty < 0 or difficulty > 10000 then
+					error_msg = "Invalid difficulty"
+				end
+			elseif k == "players" then
+				player_count = tonumber(v)
```

Comment:
```suggestion
				player_count = math.floor(tonumber(v))
```
should be an integer.

Reply:
Done.